projects/smoke_pca9539r/src/main.c

```diff
@@ -0,0 +1,123 @@
+/* Smoke test for pca9539r gpio expander. i2c and other settings based off of
+   test_pca9539r_gpio_expander.c Program sets all pins as inputs, and low state, then outputs with
+   high state, and checks configuration registers against expected results Then toggles gpio state
+   on all pins every second */
+
+#include "gpio.h"
+#include "i2c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "pca9539r_gpio_expander.h"
+#include "soft_timer.h"
+#include "test_helpers.h"
+#include "wait.h"
+
+// this used to set test mode. If set to 1, will toggle gpio states for outputs
+#define TEST_MODE_OUTPUT 1
+
+// these i2c settings may need adjustment - taken from test_pca9539r_gpio_expander.c
+#define TEST_I2C_PORT I2C_PORT_2
+#define TEST_CONFIG_PIN_I2C_SCL \
+  { GPIO_PORT_B, 10 }
+#define TEST_CONFIG_PIN_I2C_SDA \
+  { GPIO_PORT_B, 11 }
+#define VALID_I2C_ADDRESS 0x74
```

Comment:
Please rename these to something like `I2C_PORT`, `I2C_SCL_PIN`, `I2C_SDA_PIN`, and `I2C_ADDRESS` respectively and move the address beside the port. You should then give instructions to the hardware person using this on how to use these parameters.
* `I2C_ADDRESS` is the I2C address of the PCA9539R
* `I2C_PORT` is which I2C port to use (we have two; `I2C_PORT_1` has SDA on PB9 and SCL on PB8, `I2C_PORT_2` has SDA on PB11 and SCL on PB10)
* `I2C_SDA_PIN` and `I2C_SCL_PIN` are the GPIO addresses of the SDA (data) and SCL (clock) I2C pins.

Reply:
this is a good point, I've added a note about including instructions on the confluence page for smoke tests. https://uwmidsun.atlassian.net/wiki/spaces/ELEC/pages/1263927334/Smoke+Tests

Reply:
Commented in both possible configurations so that they can be easily changed